source/grid/manifold_lib.cc

```diff
@@ -186,6 +160,88 @@ SphericalManifold<dim,spacedim>::push_forward_gradient(const Point<spacedim> &sp
 }
 
 // ============================================================
+// SphericalManifold
+// ============================================================
+
+template <int dim, int spacedim>
+SphericalManifold<dim,spacedim>::SphericalManifold(const Point<spacedim> center):
+  center(center)
+{}
+
+template <int dim, int spacedim>
+Point<spacedim>
+SphericalManifold<dim,spacedim>::
+get_new_point (const Point<spacedim> &p1,
+               const Point<spacedim> &p2,
+               const double w) const
+{
+  Assert(w >=0.0 && w <= 1.0,
+         ExcMessage("w should be in the range [0.0,1.0]."));
+
+  if ( p1 == p2 ) return p1;
+
+  const Tensor<1,spacedim> v1 = p1 - center;
+  const Tensor<1,spacedim> v2 = p2 - center;
+  const double r1 = v1.norm();
+  const double r2 = v2.norm();
+
+  // Find the angle gamma described by v1 and v2:
+  const double gamma = std::acos((v1*v2)/(r1*r2));
+
+  // Find the angle sigma that correspont to archlengh equal to w
+  const double sigma = (1-w) * gamma;
+
+  // Versor with the same direction of v1
```

Comment:
Ha! I never heard of this word. Maybe replace it by "a unit vector in the direction of v1"?